oap-server/server-library/library-elasticsearch-client/src/main/java/org/apache/skywalking/library/elasticsearch/requests/factory/v6/codec/V6IndexRequestSerializer.java

```diff
@@ -38,6 +38,9 @@ public void serialize(final IndexRequest value, final JsonGenerator gen,
                 gen.writeStringField("_index", value.getIndex());
                 gen.writeStringField("_type", value.getType());
                 gen.writeStringField("_id", value.getId());
+                if (value.getRouting().isPresent()) {
+                    gen.writeStringField("routing", value.getRouting().get());
```

Comment:
Is the key `routing` or `_routing`? 
Have you tested whether the routing works? 
https://www.elastic.co/guide/en/elasticsearch/reference/current/mapping-routing-field.html